src/app/Library/CrudPanel/Traits/Columns.php

```diff
@@ -370,19 +370,21 @@ public function makeSureColumnHasNeededAttributes($column)
         $column = $this->makeSureColumnHasModel($column);
 
         // check if the column exists in the database (as a db column)
-        $columnExistsInDb = $this->hasDatabaseColumn($this->model->getTable(), $column['name']);
+        $column_exists_in_db = $this->hasDatabaseColumn($this->model->getTable(), $column['name']);
 
         // make sure column has tableColumn, orderable and searchLogic
-        $column['tableColumn'] = $column['tableColumn'] ?? $columnExistsInDb;
-        $column['orderable'] = $column['orderable'] ?? $columnExistsInDb;
-        $column['searchLogic'] = $column['searchLogic'] ?? $columnExistsInDb;
-
-        // check if it's a method on the model,
-        // that means it's a relationship
-        if (! $columnExistsInDb && method_exists($this->model, $column['name'])) {
-            $relatedModel = $this->model->{$column['name']}()->getRelated();
+        $column['tableColumn'] = $column['tableColumn'] ?? $column_exists_in_db;
+        $column['orderable'] = $column['orderable'] ?? $column_exists_in_db;
+        $column['searchLogic'] = $column['searchLogic'] ?? $column_exists_in_db;
+
+        // check if method exists in model so it's a possible relation
+        // but exclude possible matches if developer setup entity => false
+        $could_be_relation = method_exists($this->model, $column['name']) ? ! isset($column['entity']) || $column['entity'] !== false : isset($column['entity']) || $column['entity'] !== false;
```

Comment:
```suggestion
        $could_be_relation = method_exists($this->model, $column['name']) 
            ? ! isset($column['entity']) || $column['entity'] !== false 
            : isset($column['entity']) && $column['entity'] !== false;
```

There is a problem there, `||` should be `&&` in the second part of the ternary operator.
It's also better to read splitting the ternary over the next lines.